### DISCUSSION TITLE
Increasing PHP Memory Limit (even if not persistent)

### DISCUSSION THREAD

**Discussion by DaleJP** [nextcloud/all-in-one#659] [Questions] [answered]
I saw something like this asked in another thread and it seems that the answer was that some things need to remain standard to maintain the image.

I got some errors today that the PHP memory was exceeded when many users were uploading multiple files from the android apps and I'm wondering if I can change it without losing all the features I love about this build?

I would also like to be able to upload very large files. I transferred a 50GB video fine today so it seems that the 10GB limit for uploads doesn't actually matter but would like the instance to have access to all the resources it may need.

Thanks always for your help and congratulations again for a great tool.

If you can also tell me how to (or whether I need to) install ffmpeg to use the video conversion apps and any tweaks to make this as snappy as possible on a Ryzen 5900X Ubuntu server with 128GB of RAM and 30 users, I'll definitely send a beer/coffee/wheatgrass shake your way through a donation channel if you have it.

**Comment by szaimen** [comment, marked as answer]
Hello, you may change the php memory limit in the Nextcloud container in `/usr/local/etc/php/conf.d/nextcloud.ini` but this will not be persistant...

Ffmpeg is already contained in the Nextcloud container if you didn't know...

**Reply by DaleJP** [reply]
Brilliant, thank you. 

So it will revert on reboot or upgrade? I'll look into it and only do it if I see the errors again. I'm assuming my daily BorgBackup will also revert the setting back?

Ah, I'm glad FFMPEG is included. I didn't know so now I'll install the converter app.

Thank you!

**Reply by szaimen** [reply]
> So it will revert on reboot or upgrade? 

On every restart of the containers via the AIO interface (so also on upgrades)

> I'll look into it and only do it if I see the errors again. 

Alright 👍

> I'm assuming my daily BorgBackup will also revert the setting back?

Yes because it restarts the containers, see above.

**Reply by gadric** [reply]
I am currently testing AIO and was also wondering if `memory_limit` could be adjusted. 512MB seemed to be a little bit low. However, I was also wondering if more assigned memory has an effect on performance or anything?

**Reply by DaleJP** [reply]
I didn't end up trying it. I ended up just removing the mobile app from the equation because that was throwing the errors and didn't want to have another thing to remember to adjust. 

Until I can change it persistently, I'm just going to work around it. 

As long as my staff stick to the browser, it doesn't seem to cause a problem right now. 

I don't know how a larger deployment would cope, though. We have 5 users and it has trouble with a lot of simultaneous uploads.

**Reply by szaimen** [reply]
I think the memory limit is not the limiting facotr here as this value is per process... 

If you have problems with the android app, the app should be fixed instead...

**Reply by DaleJP** [reply]
I think you might be right. The problems only ever seem to arise with the Android app, so my suspicions lie there at the moment.
